test/runtests.jl

```diff
@@ -203,3 +224,31 @@ end
         @test_throws AssertionError constructorof(Rotation{Matrix{Float64}})(zeros(3,3))
     end
 end
+
+
+function funny_numbers(n)
+    types = [
+        Int128, Int16, Int32, Int64, Int8,
+        UInt128, UInt16, UInt32, UInt64, UInt8,
+        Float16, Float32, Float64,
+    ]
+    [T(true) for T in rand(types, n)]
+end
+
+@testset "inference" begin
+    @testset "Tuple n=$n" for n in [0,1,2,3,4,5,10,20,30,40]
+        t = Tuple(funny_numbers(n))
+        @test length(t) == n
+        @test getproperties(t) === t
+        @inferred getproperties(t)
```

Comment:
This is good. With `@btime` instead of `@inferred` here I get this compiling away completely up to `n=30`. `n=40` does not, and takes 3.5ns. But that's fine.